File: 1392-find-the-difference-of-two-arrays/find-the-difference-of-two-arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        nums1 = set(nums1)
        nums2 = set(nums2)

        answer0 = []
        answer1 = []
        answer = []
        
        for n in nums1:
            if n not in nums2:
                answer0.append(n)

        for n in nums2:
            if n not in nums1:
                answer1.append(n)
        
        answer.append(answer0)
        answer.append(answer1)
        return answer

File: 1392-find-the-difference-of-two-arrays/find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {

        // Convert to Array to Set to get the unique values from both the arrays.
        Set<Integer> s1 = new HashSet<Integer>();
        for (int n : nums1){
            s1.add(n);
        }
        Set<Integer> s2 = new HashSet<Integer>();
        for (int n : nums2){
            s2.add(n);
        }

        // Remove the number from set1 present in nums2 and from set2 present in nums1.
        for(int n : nums1){
            if(s2.contains(n)){
                s2.remove(n);
            }
        }
        for(int n : nums2){
            if(s1.contains(n)){
                s1.remove(n);
            }
        }

        // Create List of Lists using ArrayList to ass set1 and set2
        List<List<Integer>> ans = new ArrayList<>();
        for( int i=0; i<2; i++){
            ans.add(new ArrayList());
        }
        for ( Integer s : s1){
            ans.get(0).add(s);
        }
        for ( Integer s : s2){
            ans.get(1).add(s);
        }
        
        return ans;

    }
}